common/lib/xmodule/xmodule/x_module.py

```diff
@@ -411,7 +411,15 @@ def get_explicitly_set_fields_by_scope(self, scope=Scope.content):
         result = {}
         for field in self.fields.values():
             if field.scope == scope and field.is_set_on(self):
-                result[field.name] = field.read_json(self)
+                try:
+                    result[field.name] = field.read_json(self)
+                except TypeError as exception:
+                    exception_message = "{0}, Block-location:{1}, Field-name:{2}".format(
+                        exception.message,
```

Comment:
Would be more readable if we use keyword arguments instead of positional:
```
"{message}, Block-location:{location}, Field-name:{field_name}".format(
        message=exception.message,
        location=unicode(self.location),
        field_name=field.name
)
```